converter/src/DelphesEDM4HepConverter.cc

```diff
@@ -481,16 +482,39 @@ edm4hep::Track convertTrack(Track const* cand, const double magFieldBz)
     varOmega = cand->ErrorPT * cand->ErrorPT / cand->PT / cand->PT * trackState.omega * trackState.omega;
   }
 
-  // fill the covariance matrix. Indices on the diagonal are 0, 5,
-  // 9, 12, and 14, corresponding to D0, phi, omega, Z0 and
-  // tan(lambda) respectively. Currently Delphes doesn't provide
-  // correlations
+  // fill the covariance matrix. There is a conversion of units
+  // because the covariance matrix in delphes is with the original units 
+  // from Franco Bedeschi's code so meter and GeV
   auto& covMatrix = trackState.covMatrix;
-  covMatrix[0] = cand->ErrorD0 * cand->ErrorD0;
-  covMatrix[5] = cand->ErrorPhi * cand->ErrorPhi;
-  covMatrix[9] = varOmega;
-  covMatrix[12] = cand->ErrorDZ * cand->ErrorDZ;
-  covMatrix[14] = cand->ErrorCtgTheta * cand->ErrorCtgTheta;
+  
+  TMatrixDSym covaFB = cand->CovarianceMatrix();
+
+  double scale0 = 1.e3;
+  double scale1 = 1.;
+  //double scale2 = 1.e-3;
+  double scale2 = -2.* 1.e-3;    // CAREFUL: DELPHES USES THE HALF-CURVATURE
+  double scale3 = 1.e3;
+  double scale4 = 1.;
+
+  covMatrix[0]  = covaFB(0,0) *scale0 * scale0;
+  covMatrix[1]  = covaFB(0,1) *scale0 * scale1;
```

Comment:
Please don't forget https://github.com/key4hep/EDM4hep/pull/108, I.e, this matrix needs to be transposed

Reply:
thanks @andresailer , seems I forgot indeed, will transpose as soon as possible

Reply:
just did!